lib/core/viewer/components/img-viewer.component.spec.ts

```diff
@@ -344,11 +148,147 @@ describe('Test Img viewer component ', () => {
+            component = fixture.componentInstance;
+            component.blobFile = createFakeBlob();
+            const change = new SimpleChange(null, component.blobFile, true);
+            component.ngOnChanges({ 'blobFile': change });
+            fixture.detectChanges();
+        });
+
+        it('should update scales on zoom in', fakeAsync(() => {
+            spyOn(component, 'zoomIn').and.callThrough();
+            spyOn(component.cropper, 'zoom');
+            component.scale = 1.0;
+            tick();
+
+            component.zoomIn();
+            expect(component.scale).toBe(1.2);
+            expect(component.cropper.zoom).toHaveBeenCalledWith(0.2);
+
+            component.zoomIn();
+            expect(component.scale).toBe(1.4);
+            expect(component.cropper.zoom).toHaveBeenCalledWith(0.2);
+            }));
+
+        it('should update scales on zoom out', fakeAsync(() => {
+            spyOn(component, 'zoomOut').and.callThrough();
+            spyOn(component.cropper, 'zoom');
+            component.scale = 1.0;
+            tick();
+
+            component.zoomOut();
+            expect(component.scale).toBe(0.8);
+            expect(component.cropper.zoom).toHaveBeenCalledWith(-0.2);
+
+            component.zoomOut();
+            expect(component.scale).toBe(0.6);
+            expect(component.cropper.zoom).toHaveBeenCalledWith(-0.2);
+        }));
+
+        it('should not zoom out past 20%', fakeAsync(() => {
+            component.scale = 0.2;
+            tick();
+
+            component.zoomOut();
+            component.zoomOut();
+            component.zoomOut();
+
+            expect(component.scale).toBe(0.2);
+        }));
+
+        it('should show rotate button if not in read only mode', () => {
+            component.readOnly = false;
+            fixture.detectChanges();
+            const rotateButtonElement = element.querySelector('#viewer-rotate-button');
+            expect(rotateButtonElement).not.toEqual(null);
+        });
+
+        it('should not show rotate button by default', () => {
+            const rotateButtonElement = element.querySelector('#viewer-rotate-button');
+            expect(rotateButtonElement).toEqual(null);
+        });
+
+        it('should rotate image by -90 degrees on button click', fakeAsync(() => {
+            component.readOnly = false;
+            spyOn(component, 'rotateImage').and.callThrough();
+            spyOn(component.cropper, 'rotate');
+            fixture.detectChanges();
+            const rotateButtonElement = fixture.debugElement.query(By.css('#viewer-rotate-button'));
+            rotateButtonElement.triggerEventHandler('click', null);
+            tick();
+            expect(component.rotateImage).toHaveBeenCalled();
+            component.rotateImage();
+            expect(component.cropper.rotate).toHaveBeenCalledWith(-90);
+        }));
+
+        it('should display or not the second toolbar if in read only mode', fakeAsync(() => {
```

Comment:
These are two different tests:
1. Should display the second toolbar if not in read only mode
2. Should not display the second toolbar if in read only mode

Reply:
See previous comment: https://github.com/Alfresco/alfresco-ng2-components/pull/6958#discussion_r623893715